js/coindcx.js

```diff
@@ -0,0 +1,596 @@
+            'symbol': symbol,
+            'info': ticker,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeFloat (ticker, 'high'),
+            'low': this.safeFloat (ticker, 'low'),
+            'bid': this.safeFloat (ticker, 'bid'),
+            'bidVolume': undefined,
+            'ask': this.safeFloat (ticker, 'ask'),
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'close': last,
+            'last': last,
+            'previousClose': undefined,
+            'change': this.safeFloat (ticker, 'change_24_hour'),
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeFloat (ticker, 'volume'),
+            'quoteVolume': undefined,
+        };
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = 500, params = {}) {
+        // https://coindcx-official.github.io/rest-api/?shell#candles
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOHLCV requires a `symbol` argument');
+        }
+        await this.loadMarkets ();
+        if (!(symbol in this.markets)) {
+            throw new ExchangeError (this.id + ' has no symbol ' + symbol);
+        }
+        const market = this.market (symbol);
+        const marketInfo = this.safeValue (market, 'info');
+        const coindcxPair = this.safeString (marketInfo, 'pair');
+        if (coindcxPair === undefined) {
+            throw new ExchangeError (this.id + ' has no pair (look at market\'s info) value for ' + symbol);
+        }
+        const coindcxTimeframe = this.timeframes[timeframe];
+        if (coindcxTimeframe === undefined) {
+            throw new ExchangeError (this.id + ' has no "' + timeframe + '" timeframe');
+        }
+        const request = {
+            'pair': coindcxPair,
+            'interval': coindcxTimeframe,
+            'limit': limit,
+        };
+        const response = await this.publicGetMarketDataCandles (this.extend (request, params));
+        return this.parseOHLCVs (response, market, timeframe, since, limit);
+    }
+
+    parseOHLCV (ohlcv, market = undefined, timeframe = '1m', since = undefined, limit = undefined) {
+        return [
+            this.safeInteger (ohlcv, 'time'),
+            this.safeFloat (ohlcv, 'open'),
+            this.safeFloat (ohlcv, 'high'),
+            this.safeFloat (ohlcv, 'low'),
+            this.safeFloat (ohlcv, 'close'),
+            this.safeFloat (ohlcv, 'volume'),
+        ];
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = 30, params = {}) {
+        // https://coindcx-official.github.io/rest-api/?shell#trades
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchTrades requires a `symbol` argument');
+        }
+        await this.loadMarkets ();
+        if (!(symbol in this.markets)) {
```

Comment:
This check is not necessary here (or anywhere else in similar places), because the existence of a market is checked inside the `this.market()` call on the next line.